components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/test/java/org/wso2/carbon/identity/application/authentication/framework/config/model/graph/js/JsGraalAuthenticationContextTest.java

```diff
@@ -0,0 +1,253 @@
+        basicAuthenticatorConfig.setEnabled(true);
+        SequenceConfig sequenceConfig = new SequenceConfig();
+        Map<Integer, StepConfig> stepConfigMap = new HashMap<>();
+        StepConfig stepConfig = new StepConfig();
+        stepConfig.setOrder(1);
+        stepConfig.setAuthenticatedIdP(TEST_IDP);
+        stepConfig.setAuthenticatedAutenticator(basicAuthenticatorConfig);
+        stepConfigMap.put(1, stepConfig);
+        sequenceConfig.setStepMap(stepConfigMap);
+        AuthenticationGraph authenticationGraph = new AuthenticationGraph();
+        authenticationGraph.setStepMap(stepConfigMap);
+        sequenceConfig.setAuthenticationGraph(authenticationGraph);
+        context.setSequenceConfig(sequenceConfig);
+        Map<String, AuthenticatedIdPData> idPDataMap = new HashMap<>();
+        AuthenticatedIdPData idPData = new AuthenticatedIdPData();
+        idPData.setUser(authenticatedUser);
+        idPData.setIdpName(TEST_IDP);
+        idPDataMap.put(TEST_IDP, idPData);
+        context.setCurrentAuthenticatedIdPs(idPDataMap);
+    }
+
+    @Test
+    public void testRemoteAddition() throws IOException {
+
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        AuthenticationContext authenticationContext = new AuthenticationContext();
+        setupAuthContextWithStepData(authenticationContext, authenticatedUser);
+
+        JsGraalAuthenticationContext jsAuthenticationContext = new JsGraalAuthenticationContext(authenticationContext);
+        Value bindings = context.getBindings(POLYGLOT_LANGUAGE);
+        bindings.putMember("context", jsAuthenticationContext);
+
+        context.eval(
+                Source.newBuilder(POLYGLOT_LANGUAGE, "context.steps[1].subject.remoteClaims['testClaim']='testValue'",
+                        POLYGLOT_SOURCE).build());
+
+        ClaimMapping claimMapping = ClaimMapping.build("testClaim", "testClaim", "", false);
+        String claimCreatedByJs = authenticatedUser.getUserAttributes().get(claimMapping);
+        assertEquals(claimCreatedByJs, "testValue");
+    }
+
+    @Test
+    public void testGetServiceProviderFromWrappedContext() throws Exception {
+
+        AuthenticationContext authenticationContext = new AuthenticationContext();
+        authenticationContext.setServiceProviderName(SERVICE_PROVIDER_NAME);
+
+        JsGraalAuthenticationContext jsAuthenticationContext = new JsGraalAuthenticationContext(authenticationContext);
+        Value bindings = context.getBindings(POLYGLOT_LANGUAGE);
+        bindings.putMember("context", jsAuthenticationContext);
+
+        Value result = context.eval(
+                Source.newBuilder(POLYGLOT_LANGUAGE, "context.serviceProviderName", POLYGLOT_SOURCE).build());
+        assertFalse(result.isNull());
+        assertEquals(result.asString(), SERVICE_PROVIDER_NAME,
+                "Service Provider name set in AuthenticationContext is not " +
+                        "accessible from JSAuthenticationContext");
```

Comment:
format this properly

Reply:
Addressed with f00358d0066bbf13b65c87f4867e2ad5052761fc